mlflow/models/container/__init__.py

```diff
@@ -108,17 +73,52 @@ def _serve():
         raise Exception("This container only supports models with the MLeap or PyFunc flavors.")
 
 
+def _install_pyfunc_deps(model_path=None, install_mlflow=False):
+    """
+    Creates a conda env for serving the model at the specified path and installs almost all serving
+    dependencies into the environment - MLflow is not installed as it's not available via conda.
+    """
+    # If model is a pyfunc model, create its conda env (even if it also has mleap flavor)
+    has_env = False
+    if model_path:
+        model_config_path = os.path.join(model_path, "MLmodel")
+        model = Model.load(model_config_path)
+        # NOTE: this differs from _serve cause we always activate the env even if you're serving
+        # an mleap model
+        if pyfunc.FLAVOR_NAME not in model.flavors:
+            return
+        conf = model.flavors[pyfunc.FLAVOR_NAME]
+        if pyfunc.ENV in conf:
+            print("creating and activating custom environment")
+            env = conf[pyfunc.ENV]
+            env_path = os.path.join(model_path, env)
+            conda_create_model_env = "conda env create -n custom_env -f {}".format(env_path)
+            if Popen(["bash", "-c", conda_create_model_env]).wait() != 0:
+                raise Exception("Failed to create model environment.")
+            has_env = True
+    activate_cmd = ["source /miniconda/bin/activate custom_env"] if has_env else []
```

Comment:
Ah nice, guess activating the conda env in the container wasn't that bad after all...